src/functionality.py

```diff
@@ -89,28 +818,124 @@ async def rpactive(ctx: commands.Context):
                 await user.add_roles(rp_active_role)
                 await ctx.send(f'Added role **{guild_details.rp_active_role_name}**.')
 
+        @bot.command(
+            name='timezone',
+            brief='Lists the time based on <important> timezones.',
+            help=('Lists the time based on <important> timezones. Date format: YYYY-MM-DD.'
+                  '\nArguments: $timezone <seconds>'
+                  '\nExample: $timezone HN'
+                  '\n<seconds> is an optional field; it is an epoch/unix second argument. It can also be an RP Serial Code.'
+                  "\nInputting <seconds> as an RP Serial Code will return you said RP's First Session Date."
+                  )
+        )
+        async def timezone(ctx: commands.Context, seconds=None):
+            if not _validate_command(ctx):
+                return
 
-        async def _relay_message(message, prefix='', suffix=''):
-            for role in message.author.roles:
-                if role.id in guild_details.relaying_ignore_roles:
-                    return
+            irl_time = None
+            title_embed = None
 
-            to_channel_name = guild_details.relaying_channels[message.channel.name]
-            to_channel = _get_channel(to_channel_name)
+            if not seconds:
+                irl_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(round(time.time())))
+                title_embed = f'It is {irl_time} UTC'
 
-            user = f'{message.author.name}#{message.author.discriminator} (<@{message.author.id}>)'
+            try:
+                seconds = int(seconds)
+                irl_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(seconds))
+                title_embed = f'It is {irl_time} UTC'
+            except ValueError:
+                rp_list = rp_id_check()
+                for rp in rp_list:
+                    if seconds == rp["serial_code"]:
+                        irl_time = time.strftime('%Y-%m-%d %H:%M:%S', time.gmtime(int(rp["rp_start_time"])))
+                        title_embed = f'RP Hosted at {irl_time} UTC'
+            except TypeError as p:
+                raise p
```

Comment:
What sort of situations do you expect this `raise p` to be executed?

Reply:
Whoops, I forgot to delete that line as it was more for testing.